LOS_Driver/Motor_Channel/src/LOS_D_PWMChannel.cpp

```diff
@@ -0,0 +1,19 @@
+#include "PWMChannel.hpp"
+
+PWMChannel::PWMChannel(uint16_t pinNum, GPIO_TypeDef* port, TIM_HandleTypeDef* timer, 
+                        uint16_t timerChannel) :    pin_num_(pin_num), 
+                                                    port_(port), 
+                                                    timer_(timer), 
+                                                    timer_channel_(timer_channel) {
```

Comment:
My notes on static assertion in `config.hpp` may also be valid for these constructors as well (is that possible?). Not sure if there is some mechanism we can make us of to guarantee that they are created at compile time, and thus avoid the possibility of runtime configuration errors?

Reply:
I'm not sure I follow how this is a continuation from config.hpp. Do you mean checking the the port and timer are not nullptrs?

Reply:
I was more thinking about constexpr evaluation of the constructor, and ensuring it's available at compile time.